homeassistant/components/samsungtv/config_flow.py

```diff
@@ -78,15 +78,72 @@ def _get_entry(self):
             data=data,
         )
 
+    async def _async_set_device_unique_id(self, raise_on_progress=True):
+        """Set device unique_id."""
+
```

Comment:
```
        for progress in self._async_in_progress():
            if progress.get("context", {}).get(CONF_HOST) == self._host:
                return self.async_abort(reason="already_in_progress")

```

We should abort if the tv has already been discovered by another discovery method

We will need to set `self.context[CONF_HOST] = self._host` as well